From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Apthorp <jeremya@chromium.org>
Date: Fri, 20 Sep 2019 16:44:18 -0400
Subject: export FetchAPI mojo traits to fix component build

Without these, we get link errors in the component build when using the
blink::CloneableMessage mojo traits.

diff --git a/third_party/blink/public/common/fetch/fetch_api_request_body_mojom_traits.h b/third_party/blink/public/common/fetch/fetch_api_request_body_mojom_traits.h
index 1ddfc2108f9a0104247ea2559b597552cd20a342..f9a10b5b428e29a8824b87616f19292b38e38024 100644
--- a/third_party/blink/public/common/fetch/fetch_api_request_body_mojom_traits.h
+++ b/third_party/blink/public/common/fetch/fetch_api_request_body_mojom_traits.h
@@ -11,12 +11,13 @@
 #include "mojo/public/cpp/bindings/pending_remote.h"
 #include "services/network/public/cpp/resource_request_body.h"
 #include "services/network/public/mojom/url_loader.mojom-forward.h"
+#include "third_party/blink/public/common/common_export.h"
 #include "third_party/blink/public/mojom/fetch/fetch_api_request.mojom-forward.h"
 
 namespace mojo {
 
 template <>
-struct StructTraits<blink::mojom::FetchAPIRequestBodyDataView,
+struct BLINK_COMMON_EXPORT StructTraits<blink::mojom::FetchAPIRequestBodyDataView,
                     scoped_refptr<network::ResourceRequestBody>> {
   static bool IsNull(const scoped_refptr<network::ResourceRequestBody>& r) {
     return !r;
@@ -46,7 +47,7 @@ struct StructTraits<blink::mojom::FetchAPIRequestBodyDataView,
 };
 
 template <>
-struct StructTraits<blink::mojom::FetchAPIDataElementDataView,
+struct BLINK_COMMON_EXPORT StructTraits<blink::mojom::FetchAPIDataElementDataView,
                     network::DataElement> {
   static const network::mojom::DataElementType& type(
       const network::DataElement& element) {
